Source/WebCore/dom/Document.cpp

```diff
@@ -10911,9 +10911,9 @@ void Document::removeTopLayerElement(Element& element)
 
 HTMLDialogElement* Document::activeModalDialog() const
 {
-    for (auto& element : makeReversedRange(m_topLayerElements)) {
-        if (RefPtr dialog = dynamicDowncast<HTMLDialogElement>(element.get()); dialog && dialog->isModal())
-            return dialog.unsafeGet();
+    for (Ref element : makeReversedRange(m_topLayerElements)) {
+        if (auto* dialog = dynamicDowncast<HTMLDialogElement>(element.get()); dialog && dialog->isModal())
```

Comment:
I don't think this is safe. If element was the last `Ref` to this element, then we'd return a pointer value to freed memory. Why does element need to be `Ref` at all? It seems like everything being called here is trivial so the whole thing should be considered as trivial.

Reply:
We should be able to get it as `Ref&` as we were.